### PR TITLE
Move lock prefs to AppPreferences

### PR DESCRIPTION
**Pull request by ezaquarii**
Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### CODE REVIEW

**Comment by ezaquarii** [comment]
```
Class com.owncloud.android.utils.DisplayUtils has a circular dependency with other classes
Bug type FCCD_FIND_CLASS_CIRCULAR_DEPENDENCY (click for details) In 
class com.owncloud.android.utils.DisplayUtilsIn 
class com.owncloud.android.datamodel.ThumbnailsCacheManagerIn 
class com.owncloud.android.MainAppIn 
class com.owncloud.android.authentication.PassCodeManagerIn 
class com.owncloud.android.ui.activity.RequestCredentialsActivity
At DisplayUtils.java:[lines 105-786]

```

O gosh...

**Comment by ezaquarii** [comment]
I fixed other warning in `DisplayUtils` to decrease number of warnings back to previous value.

File: src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java [hunk 1]
```diff
@@ -83,47 +86,35 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
-                appPrefsEditor.putString(PassCodeActivity.PREFERENCE_PASSCODE_D4, passCodeDigits[3]);
+                preferences.setPassCode(
+                    passCodeDigits[0],
+                    passCodeDigits[1],
+                    passCodeDigits[2],
+                    passCodeDigits[3]
+                );
             }
 
-            appPrefsEditor.putString(SettingsActivity.PREFERENCE_LOCK, lockPref);
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java]: @ezaquarii you did remove the re-setting/enabling of the lock mechanism _after_ the preference wipe and also calculating the result based on clear+re-add.

**Comment by AndyScherzinger** [comment]
@ezaquarii looks good code wise except for one comment, see https://github.com/nextcloud/android/pull/3740#pullrequestreview-213246843

File: src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java (hunk 1 above)
Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java]: Oupsie. Will fix that.

Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java]: Fixed.

File: src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java [hunk 2]
```diff
@@ -59,10 +60,12 @@ protected void onCreate(Bundle savedInstanceState) {
         clearDataButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                ClearDataAsynTask clearDataTask = new ClearDataAsynTask();
+                ClearDataAsyncTask clearDataTask = new ClearDataAsyncTask();
                 clearDataTask.execute();
             }
         });
+
+        preferences = com.nextcloud.client.preferences.PreferenceManager.fromContext(this);
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java]: looking at the imports the full qualified class name shouldn't be needed.

**Comment by AndyScherzinger** [comment]
Just a minor comment @ezaquarii 👍

File: src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java (hunk 2 above)
Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/activity/ManageSpaceActivity.java]: Good catch. Fixed.

**Pull request merged**